x-pack/plugins/reporting/server/config/index.ts

```diff
@@ -32,42 +32,6 @@ export const config: PluginConfigDescriptor<ReportingConfigType> = {
     unused('capture.viewport', { level: 'warning' }), // deprecated as unused since 7.16
     (settings, fromPath, addDeprecation) => {
       const reporting = get(settings, fromPath);
-      if (reporting?.roles?.enabled !== false) {
```

Comment:
I'm just realizing we need a deprecation to be logged if `xpack.reporting.roles.allow` is set, because we can't seamlessly carry support for that functionality in 9.0.

I recommend restoring all the lines that are removed here, but change the condition on line 35 to:
```
      const allowedRoles = reporting?.roles?.allow ?? [];
      if (allowedRoles.length !== 0) {
```

Sorry about that!